src/app/components/registration-form/registration-form.component.ts

```diff
@@ -0,0 +1,31 @@
+import { Component } from '@angular/core';
+import { NgForm } from '@angular/forms';
+import { User } from 'src/types/user';
+import { GatewayService } from 'src/app/services/gateway.service';
+
+@Component({
+    selector: 'app-registration-form',
+    templateUrl: './registration-form.component.html',
+    styleUrls: ['./registration-form.component.scss'],
+})
+export class RegistrationFormComponent {
+    user: User = {
+        username: '',
+        email: '',
+        password: '',
+    };
+
+    constructor(private _gatewayService: GatewayService) {}
+
+    onSubmit(registrationform: NgForm) {
+        console.log('in onSubmit: ', registrationform.valid);
+        if (registrationform.valid) {
```

Comment:
A different way of doing this could be
```
if (!registrationform.valid) throw new Error("Not valid input");
this._gatewayService.postRegistrationForm(...).subscribe(...)
```